From 439da7a6f48fa965c2fe2d90b50954c609827e7b Mon Sep 17 00:00:00 2001
Message-Id: <439da7a6f48fa965c2fe2d90b50954c609827e7b.1350045285.git.minovotn@redhat.com>
In-Reply-To: <8d140e6fe679a73fd193609be44d508f17f7892b.1350045285.git.minovotn@redhat.com>
References: <8d140e6fe679a73fd193609be44d508f17f7892b.1350045285.git.minovotn@redhat.com>
From: Luiz Capitulino <lcapitulino@redhat.com>
Date: Thu, 11 Oct 2012 18:24:20 +0200
Subject: [PATCH 08/12] monitor: memory_save(): pass error message to
 OpenFileFailed error

RH-Author: Luiz Capitulino <lcapitulino@redhat.com>
Message-id: <1349979864-509-3-git-send-email-lcapitulino@redhat.com>
Patchwork-id: 42991
O-Subject: [RHEL6.4 qemu-kvm PATCH 2/6] monitor: memory_save(): pass error message to OpenFileFailed error
Bugzilla: 806775
RH-Acked-by: Laszlo Ersek <lersek@redhat.com>
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>
RH-Acked-by: Pavel Hrdina <phrdina@redhat.com>

Signed-off-by: Luiz Capitulino <lcapitulino@redhat.com>
---
 monitor.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

Signed-off-by: Michal Novotny <minovotn@redhat.com>
---
 monitor.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/monitor.c b/monitor.c
index 14a84e9..e3c2e6b 100644
--- a/monitor.c
+++ b/monitor.c
@@ -1744,7 +1744,7 @@ static int do_memory_save(Monitor *mon, const QDict *qdict, QObject **ret_data)
 
     f = fopen(filename, "wb");
     if (!f) {
-        qerror_report(QERR_OPEN_FILE_FAILED, filename, "");
+        qerror_report(QERR_OPEN_FILE_FAILED, filename, strerror(errno));
         return -1;
     }
     while (size != 0) {
@@ -1780,7 +1780,7 @@ static int do_physical_memory_save(Monitor *mon, const QDict *qdict,
 
     f = fopen(filename, "wb");
     if (!f) {
-        qerror_report(QERR_OPEN_FILE_FAILED, filename, "");
+        qerror_report(QERR_OPEN_FILE_FAILED, filename, strerror(errno));
         return -1;
     }
     while (size != 0) {
-- 
1.7.11.7

